### PR TITLE
Editor console args

### PR DESCRIPTION
**Pull request by PiotrSquaresMoscicki**

### CODE REVIEW

File: PolyEngine/Editor/Src/Configs/ProjectConfig.cpp [hunk 1]
```diff
@@ -40,19 +42,42 @@ String ProjectConfig::GetGameDllPath() const
...
 #if defined(_WIN32)
-	builder.Append(ProjectPath);
-	builder.Append("/Dist/Debug/PolyRenderingDeviceGL.dll");
+	builder.Append(GetDistDir());
+	builder.Append("PolyRenderingDeviceGL.dll");
 #elif defined(__linux__) || defined(__APPLE__)
-	builder.Append(ProjectPath)
-		.Append("/Dist/Debug/libPolyRenderingDeviceGL.so");
+	builder.Append(GetDistDir())
+		.Append("libPolyRenderingDeviceGL.so");
```
Comment (MuniuDev) [inline PolyEngine/Editor/Src/Configs/ProjectConfig.cpp]: Why? This is all in `AssetsPathConfig.json`. Extract it from there.

File: PolyEngine/Editor/Src/Configs/ProjectConfig.cpp [hunk 2]
```diff
@@ -40,19 +42,42 @@ String ProjectConfig::GetGameDllPath() const
...
+Poly::String ProjectConfig::GetDistDir() const
+{
+	StringBuilder builder;
+
+	builder.Append(ProjectDirPath);
+	builder.Append("/Dist/");
+
+	switch (ProjectConfiguration)
+	{
+	case eConfiguration::DEBUG:
```
Comment (MuniuDev) [inline PolyEngine/Editor/Src/Configs/ProjectConfig.cpp]: What about other configurations?

File: PolyEngine/Editor/Src/Widgets/Inspectors/ViewportInspectorWidget.cpp [hunk 3]
```diff
@@ -209,7 +209,7 @@ void ViewportInspectorWidget::leaveEvent(QEvent* e)
 //------------------------------------------------------------------------------
 void ViewportInspectorWidget::resizeEvent(QResizeEvent* resizeEvent)
 {
-	if (gApp->EngineMgr->GetEngineState() == PolyEditor::eEngineState::NONE)
+	if (!Manager->GetEngine())
```
Comment (MuniuDev) [inline PolyEngine/Editor/Src/Widgets/Inspectors/ViewportInspectorWidget.cpp]: This is very bad name for a field. Append some prefix, like `<sth>Manager`

**Review by MuniuDev — CHANGES_REQUESTED** [review]

File: PolyEngine/Editor/Src/Controllers/Impl/EngineController.cpp [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+void EngineController::SetEngine(std::unique_ptr<Poly::Engine> engineToControl)
+{
+	ControlledEngine = std::move(engineToControl);
+	Editor = static_cast<PolyEditor::IEditor*>(ControlledEngine->GetEditor());
+}
+
+//------------------------------------------------------------------------------
+void EngineController::Edit()
+{
+	ASSERTE(ControlledEngine, "Engine instance is requuired to perform this operation");
```
Comment (JDuchniewicz) [inline PolyEngine/Editor/Src/Controllers/Impl/EngineController.cpp]: s/requuired/required/g

File: PolyEngine/Editor/Src/EditorApp.cpp [hunk 5]
```diff
@@ -8,19 +8,38 @@
 
 EditorApp* gApp = nullptr;
 
+std::unique_ptr<ProjectConfig> GetProjectConfigFromArgs(int argc, char *argv[])
+{
+	constexpr unsigned PROJ_JSON_FILE_PATH_INDEX = 1;
+	constexpr unsigned CONFIGURSTION_INDEX = 2;
```
Comment (JDuchniewicz) [inline PolyEngine/Editor/Src/EditorApp.cpp]: s/CONFIGURSTION_INDEX/CONFIGURATION_INDEX/g

File: PolyEngine/Editor/Src/Configs/ProjectConfig.cpp (hunk 1 above)
Comment (PiotrMoscicki) [reply PolyEngine/Editor/Src/Configs/ProjectConfig.cpp]: Done

File: PolyEngine/Editor/Src/Configs/ProjectConfig.cpp (hunk 2 above)
Comment (PiotrMoscicki) [reply PolyEngine/Editor/Src/Configs/ProjectConfig.cpp]: Done

File: PolyEngine/Editor/Src/Controllers/Impl/EngineController.cpp (hunk 4 above)
Comment (PiotrMoscicki) [reply PolyEngine/Editor/Src/Controllers/Impl/EngineController.cpp]: Done

File: PolyEngine/Editor/Src/EditorApp.cpp (hunk 5 above)
Comment (PiotrMoscicki) [reply PolyEngine/Editor/Src/EditorApp.cpp]: Done

File: PolyEngine/Editor/Src/Widgets/Inspectors/ViewportInspectorWidget.cpp (hunk 3 above)
Comment (PiotrMoscicki) [reply PolyEngine/Editor/Src/Widgets/Inspectors/ViewportInspectorWidget.cpp]: This will be fixed in next PR.

**Review by MuniuDev — APPROVED** [review]

**Pull request merged**